src/Wallpaper/Portal.vala

```diff
@@ -1,14 +1,123 @@
+/* Refer to the provided link for the response codes.
+
+https://flatpak.github.io/xdg-desktop-portal/docs/doc-org.freedesktop.portal.Request.html#org-freedesktop-portal-request
+
+0: Success; the request is executed.
+1: The user cancelled the interaction.
+2: The user interaction concluded in an alternate manner.*/
+
 [DBus (name = "org.freedesktop.impl.portal.Wallpaper")]
 public class Wallpaper.Portal : Object {
     private DBusConnection connection;
+    private const string BACKGROUND_SCHEMA = "org.gnome.desktop.background";
+    private const string LOCKSCREEN_SCHEMA = "org.gnome.desktop.screensaver";
 
     public Portal (DBusConnection connection) {
         this.connection = connection;
     }
 
+
+    public void show_preview_image(string uri, string parent_window) throws GLib.Error {
+       
+        var window = new Gtk.Window();
+        window.title = "Image Preview";
+        window.set_default_size(400, 400);
+        var image = new Gtk.Image.from_file(uri);
+        image.set_size_request(400, 400);
+        var box = new Gtk.Box(Gtk.Orientation.HORIZONTAL, 0);
+        box.append(image);
+        window.set_child(box);
+        window.show();
+        
+    }
+
+
+    private void set_gsettings(string schema, string key, string value) throws Error {
+       
+        var settings = new GLib.Settings(schema);
+        settings.set_string(key, value);
+        settings.apply();
+      
+        
+    }
+
+    public void set_background_wallpaper(string uri) throws Error {
+        set_gsettings(BACKGROUND_SCHEMA, "picture-uri", uri);
+    }
+    
+    public void set_lockscreen_wallpaper(string uri) throws Error {
+        set_gsettings(LOCKSCREEN_SCHEMA, "picture-uri", uri);
```

Comment:
Also, update the `regolith.lockscreen.wallpaper.file` key in XResources followed by a reloading sway/i3 (based on the session type).